_sources/src/saltproc.rst.txt

```diff
@@ -1,34 +1,34 @@
-saltproc package
-================
-
-Subpackages
------------
-
-.. toctree::
-   :maxdepth: 4
-
-   saltproc.tests
-
-Submodules
-----------
-
-.. toctree::
-   :maxdepth: 4
-
-   saltproc.app
-   saltproc.depcode
-   saltproc.materialflow
-   saltproc.process
-   saltproc.reactor
-   saltproc.separator
-   saltproc.simulation
-   saltproc.sparger
-   saltproc.version
-
-Module contents
----------------
-
-.. automodule:: saltproc
-   :members:
-   :undoc-members:
-   :show-inheritance:
+saltproc package
```

Comment:
how did this file change? I can't see the difference here.

Reply:
> Hey, it seems like a lot more than generated HTML are in this PR (e.g. all the `_static/fonts/`) Do you know why that is?

We should add a guide somewhere on this in the near future, but for now, here's my understanding of what's going on here based on the instructions in the read me for building the documentation. The TL;DR is that the entire website gets rebuilt every time we update the documentation. I think @mehmeturkmen alluded to this in his comment. Below is the full version, for those interested.

0. We first need to build and install saltproc. If we have done this in an environment, we need to make sure that environment is active before we go build the docs. This is so `sphinc-apidoc` can read the different internal (`Depcode`, `App`, etc) and external (`pyne`, ` modules 

1. running `sphinx-apidoc --separate --force --output-dir=src/ ../saltproc` pulls the docstrings from the source code and places them in formatted `rst` files under `saltproc/src/`:
```
Creating file src/setup.rst.
Creating file src/saltproc.rst.
Creating file src/saltproc.app.rst.
Creating file src/saltproc.depcode.rst.
Creating file src/saltproc.materialflow.rst.
Creating file src/saltproc.process.rst.
Creating file src/saltproc.reactor.rst.
Creating file src/saltproc.separator.rst.
Creating file src/saltproc.simulation.rst.
Creating file src/saltproc.sparger.rst.
Creating file src/saltproc.version.rst.
Creating file src/saltproc.tests.rst.
Creating file src/saltproc.tests.test_app.rst.
Creating file src/saltproc.tests.test_depcode.rst.
Creating file src/saltproc.tests.test_materialflow.rst.
Creating file src/saltproc.tests.test_process.rst.
Creating file src/saltproc.tests.test_separator.rst.
Creating file src/saltproc.tests.test_simulation.rst.
Creating file src/saltproc.tests.test_sparger.rst.
Creating file src/modules.rst.
```
2. Running `make clean` removes any previous `_build` folder that would contain all of the files that need to go onto `gh-pages`:
```
rm -rf _build/*
```

3. Running `make html` builds the files for the web based documentation. Notice that there are several warning below due to my environment being active during the build process. This shouldn't effect the website files (I think)
```
sphinx-build -b html -d _build/doctrees   . _build/html
Running Sphinx v4.2.0
making output directory... done
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 24 source files that are out of date
updating environment: [new config] 24 added, 0 changed, 0 removed
reading sources... [100%] src/saltproc.version                                          
/home/ooblack/miniconda3/envs/saltproc-env/lib/python3.9/site-packages/saltproc-0.3.dev0-py3.9.egg/saltproc/separator.py:docstring of saltproc.separator.Separator.calc_rem_efficiency:1: WARNING: duplicate object description of saltproc.separator.Separator.calc_rem_efficiency, other instance in src/saltproc.separator, use :noindex: for one of them
/home/ooblack/miniconda3/envs/saltproc-env/lib/python3.9/site-packages/saltproc-0.3.dev0-py3.9.egg/saltproc/separator.py:docstring of saltproc.separator.Separator.description:1: WARNING: duplicate object description of saltproc.separator.Separator.description, other instance in src/saltproc.separator, use :noindex: for one of them
/home/ooblack/miniconda3/envs/saltproc-env/lib/python3.9/site-packages/saltproc-0.3.dev0-py3.9.egg/saltproc/separator.py:docstring of saltproc.separator.Separator.eff:1: WARNING: duplicate object description of saltproc.separator.Separator.eff, other instance in src/saltproc.separator, use :noindex: for one of them
/home/ooblack/miniconda3/envs/saltproc-env/lib/python3.9/site-packages/saltproc-0.3.dev0-py3.9.egg/saltproc/separator.py:docstring of saltproc.separator.Separator.k:1: WARNING: duplicate object description of saltproc.separator.Separator.k, other instance in src/saltproc.separator, use :noindex: for one of them
/home/ooblack/miniconda3/envs/saltproc-env/lib/python3.9/site-packages/saltproc-0.3.dev0-py3.9.egg/saltproc/sparger.py:docstring of saltproc.sparger.Sparger.calc_rem_efficiency:1: WARNING: duplicate object description of saltproc.sparger.Sparger.calc_rem_efficiency, other instance in src/saltproc.sparger, use :noindex: for one of them
/home/ooblack/miniconda3/envs/saltproc-env/lib/python3.9/site-packages/saltproc-0.3.dev0-py3.9.egg/saltproc/sparger.py:docstring of saltproc.sparger.Sparger.description:1: WARNING: duplicate object description of saltproc.sparger.Sparger.description, other instance in src/saltproc.sparger, use :noindex: for one of them
/home/ooblack/miniconda3/envs/saltproc-env/lib/python3.9/site-packages/saltproc-0.3.dev0-py3.9.egg/saltproc/sparger.py:docstring of saltproc.sparger.Sparger.diffusivity:1: WARNING: duplicate object description of saltproc.sparger.Sparger.diffusivity, other instance in src/saltproc.sparger, use :noindex: for one of them
/home/ooblack/miniconda3/envs/saltproc-env/lib/python3.9/site-packages/saltproc-0.3.dev0-py3.9.egg/saltproc/sparger.py:docstring of saltproc.sparger.Sparger.eff:1: WARNING: duplicate object description of saltproc.sparger.Sparger.eff, other instance in src/saltproc.sparger, use :noindex: for one of them
/home/ooblack/miniconda3/envs/saltproc-env/lib/python3.9/site-packages/saltproc-0.3.dev0-py3.9.egg/saltproc/sparger.py:docstring of saltproc.sparger.Sparger.eff:10: WARNING: Unexpected indentation.
/home/ooblack/miniconda3/envs/saltproc-env/lib/python3.9/site-packages/saltproc-0.3.dev0-py3.9.egg/saltproc/sparger.py:docstring of saltproc.sparger.Sparger.eff:11: WARNING: Block quote ends without a blank line; unexpected unindent.
/home/ooblack/miniconda3/envs/saltproc-env/lib/python3.9/site-packages/saltproc-0.3.dev0-py3.9.egg/saltproc/sparger.py:docstring of saltproc.sparger.Sparger.eps:1: WARNING: duplicate object description of saltproc.sparger.Sparger.eps, other instance in src/saltproc.sparger, use :noindex: for one of them
/home/ooblack/miniconda3/envs/saltproc-env/lib/python3.9/site-packages/saltproc-0.3.dev0-py3.9.egg/saltproc/sparger.py:docstring of saltproc.sparger.Sparger.exp_const:1: WARNING: duplicate object description of saltproc.sparger.Sparger.exp_const, other instance in src/saltproc.sparger, use :noindex: for one of them
/home/ooblack/miniconda3/envs/saltproc-env/lib/python3.9/site-packages/saltproc-0.3.dev0-py3.9.egg/saltproc/sparger.py:docstring of saltproc.sparger.Sparger.gas_const:1: WARNING: duplicate object description of saltproc.sparger.Sparger.gas_const, other instance in src/saltproc.sparger, use :noindex: for one of them
/home/ooblack/miniconda3/envs/saltproc-env/lib/python3.9/site-packages/saltproc-0.3.dev0-py3.9.egg/saltproc/sparger.py:docstring of saltproc.sparger.Sparger.h_const:1: WARNING: duplicate object description of saltproc.sparger.Sparger.h_const, other instance in src/saltproc.sparger, use :noindex: for one of them
/home/ooblack/miniconda3/envs/saltproc-env/lib/python3.9/site-packages/saltproc-0.3.dev0-py3.9.egg/saltproc/sparger.py:docstring of saltproc.sparger.Sparger.sherwood:1: WARNING: duplicate object description of saltproc.sparger.Sparger.sherwood, other instance in src/saltproc.sparger, use :noindex: for one of them
/home/ooblack/miniconda3/envs/saltproc-env/lib/python3.9/site-packages/saltproc-0.3.dev0-py3.9.egg/saltproc/sparger.py:docstring of saltproc.sparger.Sparger.temp_room:1: WARNING: duplicate object description of saltproc.sparger.Sparger.temp_room, other instance in src/saltproc.sparger, use :noindex: for one of them
looking for now-outdated files... none found
pickling environment... done
checking consistency... /home/ooblack/projects/saltproc/doc/src/modules.rst: WARNING: document isn't included in any toctree
done
preparing documents... done
writing output... [100%] src/saltproc.version                                   
generating indices... genindex py-modindex done
highlighting module code... [100%] saltproc.tests.test_sparger                  
writing additional pages... search done
copying static files... done
copying extra files... done
dumping search index in English (code: en)... done
dumping object inventory... done
build succeeded, 17 warnings.

The HTML pages are in _build/html.

Build finished. The HTML pages are in _build/html.
``` 
This is probably is why there are so many files changed.